packages/gatsby/src/schema/schema.js

```diff
@@ -299,11 +299,15 @@ const mergeTypes = ({
   parentSpan,
 }) => {
   // Only allow user or plugin owning the type to extend already existing type.
+  // Themes are also allowed to modify plugin-owned types, but not types already
+  // modified by another plugin.
   const typeOwner = typeComposer.getExtension(`plugin`)
   if (
     !plugin ||
+    plugin.name === typeOwner ||
     plugin.name === `default-site-plugin` ||
-    plugin.name === typeOwner
+    (plugin.name.startsWith(`gatsby-theme-`) &&
```

Comment:
This is a heavy handed suggestion but not a requirement, as detailed in https://github.com/gatsbyjs/gatsby/issues/16529#issuecomment-526687220

Reply:
Also this code would need to be modified to work for scoped packages like `@namespace/gatsby-theme-thing` as well